eclipse-collections/src/main/java/org/eclipse/collections/impl/bag/mutable/AbstractHashBag.java

```diff
@@ -327,6 +329,18 @@ public boolean contains(Object o)
         return this.groupByEach(function, HashBagMultimap.newMultimap());
     }
 
+    @Override
+    public MutableSet<T> asSet()
+    {
+        return UnmodifiableMutableSet.of(this.items.keySet());
```

Comment:
Since this will throw, MutableSet is the wrong type IMO. Why not leave the type non-overridden?

Reply:
@motlin just for coherency, even if Unmodifiable, it's still a `MutableSet` in terms of typing... but I can leave `SetIterable` here yes. I will let you decide